### PR TITLE
Extend check_dns.pl so it can take a list of domains

### PR DESCRIPTION
**Pull request by emiljp**
Extend check_dns.pl so it can take a list of domains that each have the same expected results.

I have many servers with even more websites, some servers have many websites. I want a single check for each server that checks all the websites on that server.

### CODE REVIEW

File: check_dns.pl [hunk 1]
```diff
@@ -128,101 +136,104 @@ BEGIN
...
+    } else {
+        $msg .= "OK return '";
+    }
+    if ($no_uniq_results){
+        $msg .= join("','", @results);
+    } else {
+        $msg .= join("','", @results_uniq);
+    }
+    $msg .= "'";
+    $msg .= " | dns_query_time='${call_time}s'" if $verbose;
```
Comment (HariSekhon) [inline check_dns.pl:231]: this will mess up in --verbose mode

File: check_dns.pl [hunk 2]
```diff
@@ -128,101 +136,104 @@ BEGIN
...
-    $msg .= join("','", @results);
-} else {
-    $msg .= join("','", @results_uniq);
-}
-$msg .= "'";
-$msg .= " in $total_time secs" if $verbose;
+
+my $stop_calls = time;
+my $total_time =  sprintf("%.4f", $stop_calls - $start_calls);
+my $perfdata = "\n | total_dns_query_time='${total_time}s'";
```
Comment (HariSekhon) [inline check_dns.pl:236]: multi-line will break some monitoring servers output

File: check_dns.pl [hunk 3]
```diff
@@ -128,101 +136,104 @@ BEGIN
...
-    foreach my $result (@results){
-        $result =~ /^$expected_regex2$/ or push(@regex_mismatches, $result);
+    my @regex_mismatches;
+    if($valid_expected_regex){
+        foreach my $result (@results){
+            $result =~ /^$valid_expected_regex$/ or push(@regex_mismatches, $result);
+        }
     }
+    @regex_mismatches = sort(uniq_array(@regex_mismatches)) if(@regex_mismatches);
+    $msg .= "\n$records[$i] $type record ";
```
Comment (HariSekhon) [inline check_dns.pl:213]: multi-line will break some monitoring servers output / dashboards as traditionally only 1 line is guaranteed to be displayed, you wouldn't want the output on some systems to be "DNS OK:" and nothing else...

**Comment by HariSekhon** [comment]
Thanks for the pull request.

Overall the changes look ok, just needs some tweaking of the output formatting to maintain 1 line dashboard compatibility, verbose mode testing etc.

**Comment by emiljp** [comment]
Thanks for the feedback. I will change it to only output one line and remake this pull request from a different account. I used this account by mistake.

**Comment by EHP4321** [comment]
This is my new account. Here is the new pull request https://github.com/HariSekhon/Nagios-Plugins/pull/339

**Comment by EHP4321** [comment]
I am sorry for the bother of creating a new request instead of just committing the fixes here. But I am in the process of de-googling and is trying to remember not to use my old google accounts eveywhere.

**Pull request closed without merging**